[CI] FlattenedFieldRootBlockLoaderTests class failing

**Build Scans:**
- [elasticsearch-intake #40037 / part1](https://gradle-enterprise.elastic.co/s/xyyfjv2ubc72k)
- [elasticsearch-pull-request #147877 / part-1](https://gradle-enterprise.elastic.co/s/bxw4x6s3eqq6a)
- [elasticsearch-pull-request #147870 / part-1](https://gradle-enterprise.elastic.co/s/nrkku6onfzisk)
- [elasticsearch-pull-request #147870 / part-1](https://gradle-enterprise.elastic.co/s/iact5flsrvpzi)
- [elasticsearch-pull-request #147852 / part-1](https://gradle-enterprise.elastic.co/s/siwe66apqyv5o)
- [elasticsearch-pull-request #147852 / part-1](https://gradle-enterprise.elastic.co/s/qj3mr6mybs26m)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.mapper.blockloader.FlattenedFieldRootBlockLoaderTests.testBlockLoaderWithCranky {params=Params[syntheticSource=true, preference=NONE, binaryDocValues=true]}" -Dtests.seed=DBDCFD83E8E1E443 -Dtests.locale=bs-Cyrl -Dtests.timezone=Europe/Zaporozhye -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.index.mapper.blockloader.FlattenedFieldRootBlockLoaderTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('org.elasticsearch.index.mapper.blockloader.FlattenedFieldRootBlockLoaderTests'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<[{"ɖʔɲʊɼʕʝɳɑɬ":"0.8063690293884729","ٯقۘ.ܶ܄ܡܕܥܫ݉":"4672412782810884408","ٯقۘ.๷ฯ๣฼ฆฆ":null,"ٯقۘ.ᜡᜱᜰ᜺ᜢ᜶᜿":"ᱜᱻᱹ᱙ᱩ᱒ᱝᱻᱷᱫᱣᱸ","ٯقۘ.⟮":["ぷげ","ぷげ","꧔ꦏ꧀ꦋꦍꦸ꧌ꦯꦌ꧁ꦞꦣꦫꦼ꧎ꦥ꧊ꦋꦌꦜ꧈ꦪ꧟꧓ꦰ꧛꧟ꦔ꧕ꦭꦂꧏꦄ꧕ꦊ","ぷげ","ᱜᱻᱹ᱙ᱩ᱒ᱝᱻᱷᱫᱣᱸ"],"ٯقۘ.⡵⠐⢎⢎⡶⡎⠞⡩⣣":"","ٯقۘ.ꢦ꣗ꢉꢶꢫꢝ.న":null,"ٯقۘ.ꢦ꣗ꢉꢶꢫꢝ.༔ྍ༼྇ྨ཮ཞ༉࿅":"ⴇⴭⴌ⴯ⴏ⴫ⴚⴎⴒⴠⴈⴆⴘⴆⴎⴆ⴪ⴣ","ٯقۘ.ꢦ꣗ꢉꢶꢫꢝ.":"0.2820321953328465","ٯقۘ.ꢦ꣗ꢉꢶꢫꢝ.⺠⻤⻀⺛⺼⺦":"370245352733217637","ٯقۘ.ꢦ꣗ꢉꢶꢫꢝ.ꜥꟁꟲꞯ":["0.5263081560223846","0.5263081560223846"],"ٯقۘ.ꢦ꣗ꢉꢶꢫꢝ.勉狀辶愈舘器":"0.2820321953328465","ᵩ":["","ܠܔ܄ܜ܉ݎݎ܍݈ܷ݃ܧܬ܆ܭ݌","ힳퟡힰퟷퟬퟆퟷퟒힹퟩힸퟛ퟊ퟕ퟇퟇ퟝퟡힷힶힰퟟퟩힱퟦힽ"],"乯证峯穆":["4672412782810884408","-6708403607153414620","-6708403607153414620","-4536078289730741599"],"ꢛ":"4672412782810884408","":"0.8493209862219951"}]> but was:<[{"ɖʔɲʊɼʕʝɳɑɬ":"0.8063690293884729","ٯقۘ.ܶ܄ܡܕܥܫ݉":"4672412782810884408","ٯقۘ.๷ฯ๣฼ฆฆ":"GmASXZv","ٯقۘ.ᜡᜱᜰ᜺ᜢ᜶᜿":"ᱜᱻᱹ᱙ᱩ᱒ᱝᱻᱷᱫᱣᱸ","ٯقۘ.⟮":["ぷげ","ぷげ","꧔ꦏ꧀ꦋꦍꦸ꧌ꦯꦌ꧁ꦞꦣꦫꦼ꧎ꦥ꧊ꦋꦌꦜ꧈ꦪ꧟꧓ꦰ꧛꧟ꦔ꧕ꦭꦂꧏꦄ꧕ꦊ","ぷげ","ᱜᱻᱹ᱙ᱩ᱒ᱝᱻᱷᱫᱣᱸ"],"ٯقۘ.⡵⠐⢎⢎⡶⡎⠞⡩⣣":"","ٯقۘ.ꢦ꣗ꢉꢶꢫꢝ.న":"GmASXZv","ٯقۘ.ꢦ꣗ꢉꢶꢫꢝ.༔ྍ༼྇ྨ
[truncated]
```

**Issue Reasons:**
- [main] 2 consecutive failures in class org.elasticsearch.index.mapper.blockloader.FlattenedFieldRootBlockLoaderTests
- [main] 1 consecutive failures in step part1
- [main] 1 consecutive failures in step part-1
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 1 consecutive failures in pipeline elasticsearch-pull-request
- [main] 6 failures in class org.elasticsearch.index.mapper.blockloader.FlattenedFieldRootBlockLoaderTests (1.3% fail rate in 462 executions)
- [main] 5 failures in step part-1 (1.5% fail rate in 324 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (0.9% fail rate in 318 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -590,15 +590,3 @@ tests:
 - class: org.elasticsearch.xpack.esql.expression.function.grouping.BucketColumnMetadataIT
   method: testStatsWithMultiBucket
   issue: https://github.com/elastic/elasticsearch/issues/149595
-- class: org.elasticsearch.index.mapper.blockloader.FlattenedFieldRootBlockLoaderTests
-  method: testBlockLoaderForFieldInObject {params=Params[syntheticSource=false, preference=STORED, binaryDocValues=true]}
-  issue: https://github.com/elastic/elasticsearch/issues/149601
-- class: org.elasticsearch.index.mapper.blockloader.FlattenedFieldRootBlockLoaderTests
-  method: testBlockLoaderForFieldInObjectWithCranky {params=Params[syntheticSource=false, preference=STORED, binaryDocValues=false]}
-  issue: https://github.com/elastic/elasticsearch/issues/149602
-- class: org.elasticsearch.index.mapper.blockloader.FlattenedFieldRootBlockLoaderTests
-  method: testBlockLoaderForFieldInObject {params=Params[syntheticSource=false, preference=STORED, binaryDocValues=false]}
-  issue: https://github.com/elastic/elasticsearch/issues/149603
-- class: org.elasticsearch.index.mapper.blockloader.FlattenedFieldRootBlockLoaderTests
-  method: testBlockLoaderForFieldInObjectWithCranky {params=Params[syntheticSource=false, preference=STORED, binaryDocValues=true]}
-  issue: https://github.com/elastic/elasticsearch/issues/149604

--- a/server/src/test/java/org/elasticsearch/index/mapper/blockloader/FlattenedFieldRootBlockLoaderTests.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/blockloader/FlattenedFieldRootBlockLoaderTests.java
@@ -15,7 +15,6 @@ import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.datageneration.Mapping;
 import org.elasticsearch.index.mapper.BinaryDVBlockLoaderTestCase;
 import org.elasticsearch.index.mapper.BlockLoaderTestRunner;
-import org.elasticsearch.index.mapper.MappedFieldType;
 import org.elasticsearch.xcontent.XContentBuilder;
 import org.elasticsearch.xcontent.XContentFactory;
 
@@ -103,16 +102,10 @@ public class FlattenedFieldRootBlockLoaderTests extends BinaryDVBlockLoaderTestC
     @Override
     protected Object expected(Map<String, Object> fieldMapping, Object value, TestContext testContext) {
         var nullValue = (String) fieldMapping.get("null_value");
-        // null_value is only applied during indexing into doc values. When the block loader
-        // uses the source path (syntheticSource=false with STORED preference or no doc values),
-        // null leaf values are simply dropped.
-        boolean useDocValues = hasDocValues(fieldMapping, true);
-        if (params.syntheticSource() == false) {
-            useDocValues = useDocValues
-                && fieldMapping.get("ignore_above") == null
-                && params.preference() != MappedFieldType.FieldExtractPreference.STORED;
-        }
-        if (nullValue != null && useDocValues) {
+        // null_value is applied via doc values (substituted at index time by FlattenedFieldParser)
+        // and via the source path (substituted at read time by FlattenedSourceValueFetcher).
+        // Apply it unconditionally here to match both code paths.
+        if (nullValue != null) {
             value = applyFlattenedNullValue(value, nullValue);
         }
         ValuesMode mode = ValuesMode.from(fieldMapping, params);

